lib/srv/regular/sshserver.go

```diff
@@ -1047,6 +1059,60 @@ func (s *Server) getServerResource() (types.Resource, error) {
 	return s.getServerInfo(), nil
 }
 
+func (s *Server) filterExistingNodes(instances server.EC2Instances) (server.EC2Instances, error) {
+	var filtered []*ec2.Instance
+	nodes, err := s.auth.GetNodes(s.ctx, s.getNamespace())
+	if err != nil {
+		return server.EC2Instances{}, err
+	}
+	for _, node := range nodes {
+		for _, inst := range instances.Instances {
+			result := types.MatchLabels(node, map[string]string{
+				types.AWSAccountIDLabel:  instances.AccountID,
+				types.AWSInstanceIDLabel: aws.StringValue(inst.InstanceId),
+			})
+			if result {
+				filtered = append(filtered, inst)
+			}
+		}
+	}
+	instances.Instances = filtered
+	return instances, nil
+}
+
+func (s *Server) handleEC2Discovery() {
+	go s.cloudWatcher.Run()
+	for {
+		select {
+		case instances := <-s.cloudWatcher.InstancesC:
+			client, err := s.clients.GetAWSSSMClient(instances.Region)
+			if err != nil {
+				log.Error("error getting AWS SSM client: ", err)
+				return
+			}
+			filteredInstances, err := s.filterExistingNodes(instances)
+			if err != nil {
+				log.Errorf("Error filtering existing nodes: %s", err)
+			}
+			installer := server.NewSSMInstaller(
+				server.SSMInstallerConfig{
+					SSM:       client,
+					Instances: filteredInstances.Instances,
+					Params:    filteredInstances.Parameters,
+					Emitter:   s,
+					Ctx:       s.ctx,
+					Region:    filteredInstances.Region,
+					AccountID: filteredInstances.AccountID,
+				})
+			if err := installer.Run(filteredInstances.Document); err != nil {
```

Comment:
I think I asked before, but it's a little confusing interface that SSM installer is initialized with instances, context, etc. Those should be parameters to Run.

Can we instantiate SSMInstaller once (make it a part of Server for example) with SSM client and Emitter and make other fields parameters to Run method?

```go
type SSMInstallerConfig struct {
  SSM ...
  Emitter ...
}

type SSMRunRequest struct {
  Instances ...
  Params ...
  Region string
  AccountID string
}

func (i *ssmInstaller) Run(ctx context.Context, req SSMRunRequest) error {
  ...
}
```